src/main/ruby/truffleruby/core/random.rb

```diff
@@ -66,10 +66,18 @@ def random(limit)
   end
 
   def random_range(limit)
-    min, max = limit.max.coerce(limit.min)
+    return nil if Comparable.compare_int(limit.end <=> limit.begin) < 0
+    max = -> do
+      return limit.end unless limit.exclude_end?
+
+      limit.end - 1
+    end.call
+
+    min, max = max.coerce(limit.min)
+
     diff = max - min
     diff += 1 if max.kind_of?(Integer)
-    random(diff) + min
+    diff == 0 ? max : random(diff) + min
```

Comment:
not sure if it's safe to compare diff to 0 for the cases when max and min are floats

Reply:
`diff == 0 ? min : min + random(diff)` seems more natural (makes it clear it's to avoid `random(0)`).